sdk/search/Azure.Search.Documents/tests/Utilities/SearchTestBase.cs

```diff
@@ -287,7 +287,15 @@ static void AssertEqual(object e, object a, string path)
                 }
                 await DelayAsync(delay);
             }
-            Assert.Fail($"Indexing only reached {count} documents and not the expected {minimumCount}!");
+
+            if (count == 0)
+            {
+                Assert.Inconclusive("Indexing failed to start.");
+            }
+            else
+            {
+                Assert.Fail($"Indexing only reached {count} documents and not the expected {minimumCount}!");
+            }
```

Comment:
I think Heath's suggestion was more along the lines of

```suggestion
            Assert.Inconclusive($"Indexing only reached {count} documents and not the expected {minimumCount}!");
```

so we stop test runs from failing when the service is under enough load that our timeouts trip.

Reply:
When discussing this issue on another thread, I understood this as his suggestion.
Mark it inconclusive if indexing hasn't even started, fail if we get some results as that is a more interesting scenario.

@heaths , should we use a simplified check for all cases?

Reply:
Yeah, but with 10 retry attempts maybe my original suggestion (what Mohit implemented) is better. I still question whether we even care about the exact number in most cases - some, like batching, probably. It's just that the service is eventually consistent, which doesn't work great for testing. It'd be nice if we could divorce ourselves from that, but that now-10 retry attempts should help.

Given that, I'm in favor of keeping what Mohit currently has implemented.